### PR TITLE
DAOS-2169 vos: Add single entry estimate to overhead tool

### PR DESCRIPTION
**Pull request by jolivier23**
To get an idea of what tree optimization will do, add single
entry size to estimator.  Also, add ability to specify that
a key is metadata rather than user data for such things
as array and posix file metadata keys.  Adds SCM size
and fixes a couple of bugs with calculations

### CODE REVIEW

File: src/vos/vos_overhead.c [hunk 1]
```diff
@@ -11,6 +11,12 @@ vos_pool_get_msize(void)
 	return sizeof(struct vos_pool_df);
 }
 
+int
+vos_pool_get_scm_cutoff(void)
```
Comment (mchaarawi) [inline src/vos/vos_overhead.c]: is this really necessary? can't use VOS_BLK_SZ in tests?

File: src/vos/tests/parse_csv.py [hunk 2]
```diff
@@ -0,0 +1,261 @@
...
+    parser.add_argument('--output', dest="output", type=str,
+                        help='Output file name', default="output.yaml")
+    args = parser.parse_args()
+
+    if args.io_size > args.chunk_size:
+        print "Assuming --io-size set to --chunk-size %d\n" % args.chunk_size
+        args.io_size = args.chunk_size
+
+    if args.chunk_size % args.io_size != 0:
+        print "--chunk-size must be evenly divisble by --io-size"
```
Comment (mchaarawi) [inline src/vos/tests/parse_csv.py]: is this just a requirement for the tool to estimate properly?

File: src/vos/vos_overhead.c (hunk 1 above)
Comment (jolivier23) [reply src/vos/vos_overhead.c]: I suppose we could

File: src/vos/tests/parse_csv.py (hunk 2 above)
Comment (jolivier23) [reply src/vos/tests/parse_csv.py]: Yeah, I think it just makes the math a bit simpler that way.  Is it not generally the case?

Comment (mchaarawi) [reply src/vos/tests/parse_csv.py]: no doesn't have to be. applications can issue I/Os with any size regardless of the chunk size.
but for this tool and estimation, it's is an OK assumption.

**Pull request merged**